src/core/unit.c

```diff
@@ -2997,6 +2999,18 @@ int unit_deserialize(Unit *u, FILE *f, FDSet *fds) {
 
                         continue;
 
+                } else if (streq(l, "cgroup-realized-mask")) {
+
+                        if (safe_atoi(v, (int *)&u->cgroup_realized_mask) < 0)
+                                log_unit_debug(u, "Failed to parse cgroup-realized-mask %s, ignoring.", v);
+                        continue;
+
+                } else if (streq(l, "cgroup-enabled-mask")) {
+
+                        if (safe_atoi(v, (int *)&u->cgroup_enabled_mask) < 0)
```

Comment:
@fbuihuu i am pretty sure that if you define an enum with values > 2^32, some 64bit integer type is used instead